### PR TITLE
Merge semver to dev

### PR DESCRIPTION
**Pull request by chillaq**

### CODE REVIEW

File: client/src/main/java/io/split/engine/matchers/Semver.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            return 1;
+        }
+        // Compare pre-release versions lexically
+        int minLength = Math.min(_preRelease.length, toCompare.PreRelease().length);
+        for (int i = 0; i < minLength; i++) {
+            if (_preRelease[i].equals(toCompare.PreRelease()[i])) {
+                continue;
+            }
+            if ( Ints.tryParse(_preRelease[i]) != null &&  Ints.tryParse(toCompare._preRelease[i]) != null) {
+                return Integer.compare(Integer.parseInt(_preRelease[i]), Integer.parseInt(toCompare._preRelease[i]));
```
Comment (sanzmauro) [inline client/src/main/java/io/split/engine/matchers/Semver.java:100]: Please use Long numerics. eg: `Long.parseLong` and `Long.compare`

File: client/src/main/java/io/split/engine/matchers/EqualToSemverMatcher.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    private final Semver _semVer;
+
+    public EqualToSemverMatcher(String semVer) {
+        _semVer = Semver.build(semVer);
+    }
+
+    @Override
+    public boolean match(Object matchValue, String bucketingKey, Map<String, Object> attributes, EvaluationContext evaluationContext) {
+        if (matchValue == null || _semVer == null) {
```
[the remark this answers is not in the document]
Comment (nmayorsplit) [reply client/src/main/java/io/split/engine/matchers/EqualToSemverMatcher.java:17]: Updated.

File: client/src/main/java/io/split/engine/matchers/InListSemverMatcher.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            Semver semver = Semver.build(item);
+            if (semver == null) continue;
+
+            _semverlist.add(semver);
+        }
+    }
+
+    @Override
+    public boolean match(Object matchValue, String bucketingKey, Map<String, Object> attributes, EvaluationContext evaluationContext) {
+        if (matchValue == null || _semverlist.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (nmayorsplit) [reply client/src/main/java/io/split/engine/matchers/InListSemverMatcher.java]: Updated.

File: client/src/main/java/io/split/engine/matchers/Semver.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        } else if (_isStable && !toCompare.IsStable()) {
+            return 1;
+        }
+        // Compare pre-release versions lexically
+        int minLength = Math.min(_preRelease.length, toCompare.PreRelease().length);
+        for (int i = 0; i < minLength; i++) {
+            if (_preRelease[i].equals(toCompare.PreRelease()[i])) {
+                continue;
+            }
+            if ( Ints.tryParse(_preRelease[i]) != null &&  Ints.tryParse(toCompare._preRelease[i]) != null) {
```
[the remark this answers is not in the document]
Comment (nmayorsplit) [reply client/src/main/java/io/split/engine/matchers/Semver.java]: Updated.

**Review by mmelograno — APPROVED** [review]

**Pull request merged**